src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -395,7 +397,7 @@ public PooledObject<LinkedBuffer> wrap(LinkedBuffer obj) {
             .put(StatNames.MODEL_COUNT.getName(), new ADStat<>(false, new ModelsOnNodeCountSupplier(modelManager, cacheProvider)))
             .build();
 
-        ADStats adStats = new ADStats(stats);
+        adStats = new ADStats(stats);
```

Comment:
Per previous comment, you can leave this as a local variable that you inject, no need to make it an instance variable.

Reply:
ACk